regression-test/suites/query_p0/sql_functions/cast_function/test_cast_string_to_array.groovy

```diff
@@ -41,5 +41,14 @@ suite("test_cast_string_to_array") {
     qt_sql """ select cast ("[1,2,3,,,]" as array<int>) """
     qt_sql """ select cast ("[a,b,c,,,]" as array<string>) """
     qt_sql """ select cast ("[1.34,2.01,,,]" as array<decimal(10, 3)>) """
+
+    sql """ ADMIN SET FRONTEND CONFIG ("enable_date_conversion" = "false"); """
+    qt_sql """ select cast ("[2022-09-01,,]" as array<date>) """
+    qt_sql """ select cast ("[2022-09-01,,]" as array<string>) """
+    qt_sql """ select cast(cast ("[2022-09-01,,]" as array<string>) as array<date>) """
+
+    sql """ ADMIN SET FRONTEND CONFIG ("enable_date_conversion" = "true"); """
     qt_sql """ select cast ("[2022-09-01,,]" as array<date>) """
```

Comment:
what's the difference between [2022-09-01, , ]  and [2022-09-01, NULL, NULL]  ?

Reply:
We and doris cast empty string to null date:
```
mysql [(none)]>select cast('' as date);
+--------------------+
| cast('' as DATEV2) |
+--------------------+
| NULL               |
+--------------------+
1 row in set (0.01 sec)
```

The casting result `[2022-09-01, , ]` should be not correct, I'll take a look.